modules/javafx.graphics/src/main/java/javafx/css/CssParser.java

```diff
@@ -3835,6 +3930,196 @@ private ParsedValueImpl<ParsedValue[],Font> parseFont(Term root) throws ParseExc
         return new ParsedValueImpl<>(values, FontConverter.getInstance());
     }
 
+    // https://www.w3.org/TR/css-transitions-1/#transition-shorthand-property
+    private ParsedValueImpl<ParsedValue<ParsedValue[], TransitionDefinition>[], TransitionDefinition[]>
+            parseTransitionLayers(Term term) throws ParseException {
+        int nLayers = numberOfLayers(term);
+        ParsedValue<ParsedValue[], TransitionDefinition>[] layers = new ParsedValue[nLayers];
+
+        for (int i = 0; i < nLayers; ++i) {
+            layers[i] = parseTransition(term);
+            term = nextLayer(term);
+        }
+
+        return new ParsedValueImpl<ParsedValue<ParsedValue[], TransitionDefinition>[], TransitionDefinition[]>(
+            layers, TransitionDefinitionConverter.SequenceConverter.getInstance());
+    }
+
+    private ParsedValueImpl<ParsedValue[], TransitionDefinition> parseTransition(Term term)
+            throws ParseException{
+        ParsedValue<?, String> parsedProperty = null;
+        ParsedValue<ParsedValue<?, Size>, Duration> parsedDuration = null;
+        ParsedValue<ParsedValue<?, Size>, Duration> parsedDelay = null;
+        ParsedValue<?, Interpolator> parsedTimingFunction = null;
+
+        for (int i = 0; i < 4; ++i) {
+            if (term == null) {
+                break;
+            } else if (isEasingFunction(term.token)) {
```

Comment:
minor: No need for `else`
```suggestion
            }
            
            if (isEasingFunction(term.token)) {
```